cli/azd/pkg/pipeline/pipeline_manager.go

```diff
@@ -138,28 +147,93 @@ func (pm *PipelineManager) Configure(ctx context.Context) (result *PipelineConfi
 		return result, fmt.Errorf("ensuring git remote: %w", err)
 	}
 
-	// *********** Create or update Azure Principal ***********
-	if pm.args.PipelineServicePrincipalName == "" {
-		// This format matches what the `az` cli uses when a name is not provided, with the prefix
-		// changed from "az-cli" to "az-dev"
-		pm.args.PipelineServicePrincipalName = fmt.Sprintf("az-dev-%s", time.Now().UTC().Format("01-02-2006-15-04-05"))
+	// Existing Service Principal Lookup strategy
+	// 1. --principal-id
+	// 2. --principal-name
+	// 3. AZURE_PIPELINE_CLIENT_ID environment variable
+	// 4. Create new service principal with default naming convention
+	envClientId := pm.env.Getenv(AzurePipelineClientIdEnvVarName)
+	var appIdOrName string
+	var lookupKind servicePrincipalLookupKind
+	if pm.args.PipelineServicePrincipalId != "" {
+		appIdOrName = pm.args.PipelineServicePrincipalId
+		lookupKind = lookupKindPrincipalId
+	}
+	if appIdOrName == "" && pm.args.PipelineServicePrincipalName != "" {
+		appIdOrName = pm.args.PipelineServicePrincipalName
+		lookupKind = lookupKindPrincipleName
+	}
+	if appIdOrName == "" && envClientId != "" {
+		appIdOrName = envClientId
+		lookupKind = lookupKindEnvironmentVariable
+	}
```

Comment:
consider moving this to a single method where we can return the id as soon as possible, based on priority, like

func appIdOrName() (string, string) {
      if pm.args.PipelineServicePrincipalId != "" {
		return pm.args.PipelineServicePrincipalId, lookupKindPrincipalId
	}
...
...
...
}

Reply:
It would only be used in this one place so I don't see the value in moving it at this time.

Reply:
Well, not because who is using this code, but to avoid increasing that much the current method.
Also to be able to test it directly, calling it with all alternatives to cover all code.

And also to delegate the default name (`fmt.Sprintf("az-dev-%s", time.Now().UTC().Format("01-02-2006-15-04-05"))`) to this new method.

But yeah, it's not required for the PR.  I have some pending items to add code coverage to all `pipeline config`. I think I can do that refactoring as part of that.